### PR TITLE
Handle 401 status code (and IOException)

### PR DESCRIPTION
**Pull request by palyuga**

### CODE REVIEW

File: src/main/kotlin/bitbucket/BitbucketClient.kt
Comment (nikitapodsh) [inline src/main/kotlin/bitbucket/BitbucketClient.kt]: Logger isn't used

File: src/main/kotlin/bitbucket/BitbucketClient.kt
Comment (nikitapodsh) [inline src/main/kotlin/bitbucket/BitbucketClient.kt:26]: The BitbucketClient constructor is getting too big. I think, we could create `responseHandler` in `BitbucketClientFactory`. It wouldn't reduce not so much lines, but we would keep it fit. 

Also it would be great (may be in some other PR) to use `Settings`  from `BitbucketHelperConfigurable` or other data class. It would substitute 4 params in `BitbucketClient` for 1

File: src/main/kotlin/bitbucket/BitbucketClientFactory.kt
Comment (nikitapodsh) [inline src/main/kotlin/bitbucket/BitbucketClientFactory.kt:29]: After each invocation `BitbucketClientFactory#createClient` we create new `HttpClient` and `ObjectMapper`. We could replace it in companion object to make it static

**Review by nikitap492 — APPROVED** [review]
I've left some comments but they are more optional than required for this PR

**Comment by palyuga** [comment]
@nikitap492, thanks, I updated this PR according to your suggestions

File: src/main/kotlin/bitbucket/BitbucketClient.kt
Comment (palyuga) [reply src/main/kotlin/bitbucket/BitbucketClient.kt:26]: done

File: src/main/kotlin/bitbucket/BitbucketClient.kt
Comment (palyuga) [reply src/main/kotlin/bitbucket/BitbucketClient.kt]: removed

File: src/main/kotlin/bitbucket/BitbucketClientFactory.kt
Comment (palyuga) [reply src/main/kotlin/bitbucket/BitbucketClientFactory.kt:29]: And this one is left to do in future, I was to lazy :)

**Comment by palyuga** [comment]
@nikitap492 , could you re-approve? Your previous approve disappeared after the post-review changes,

**Review by nikitap492 — APPROVED** [review]

**Comment by nikitap492** [comment]
It looks like I don't write access and my approve isn't enough

**Comment by palyuga** [comment]
#24 was addressed here

**Pull request merged**
